sql: TestExplainRedact failed

sql.TestExplainRedact [failed](https://tanzanite.cluster.engflow.com/invocations/default/0ef52661-7385-4131-8f9e-7addef618805?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [59b261a579cbe2c032a5dd3e182ff67aeee900b9](https://github.com/cockroachdb/cockroach/commits/59b261a579cbe2c032a5dd3e182ff67aeee900b9):

Fatal error:

```
panic: test timed out after 1h16m37s
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:476 +0x69d

```
Stack: 

```
goroutine 173223 [select, 1 minutes]:
github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts/ptcache.(*Cache).periodicallyRefreshProtectedtsCache(0xc002fa2160, {0xe274408, 0xc0075715f0})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts/ptcache/cache.go:180 +0x357
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:485 +0x263
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 172976
running tests:
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:476 +0x69d
	TestExplainRedact (1h15m20s)
```
<details><summary>Log preceding fatal error</summary>
<p>

```
goroutine 173144 [sync.Cond.Wait]:
sync.runtime_notifyListWait(0xc00a4b4850, 0x181c)
	GOROOT/src/runtime/sema.go:569 +0x159
sync.(*Cond).Wait(0xc00a4b4840)
	GOROOT/src/sync/cond.go:70 +0x75
github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed.(*schedulerShard).processEvents(0xc008e89440, {0xe274408, 0xc00664d200})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed/scheduler.go:490 +0x7d6
github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed.(*Scheduler).Start.func1({0xe274408, 0xc00664d200})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed/scheduler.go:263 +0x1d2
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:485 +0x263
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 173114
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:476 +0x69d

goroutine 173562 [select]:
github.com/cockroachdb/cockroach/pkg/sql/sqlstats/insights.(*concurrentBufferIngester).Start.func1({0xc005575408?, 0xc001bef200?})
	github.com/cockroachdb/cockroach/pkg/sql/sqlstats/insights/pkg/sql/sqlstats/insights/ingester.go:97 +0x145
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:485 +0x263
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 172976
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:476 +0x69d

goroutine 174403 [select]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0xc00a3d80f0, 0x1)
	google.golang.org/grpc/internal/transport/external/org_golang_google_grpc/internal/transport/controlbuf.go:416 +0x1af
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0xc008674660)
	google.golang.org/grpc/internal/transport/external/org_golang_google_grpc/internal/transport/controlbuf.go:534 +0x174
google.golang.org/grpc/internal/transport.NewServerTransport.func2()
	google.golang.org/grpc/internal/transport/external/org_golang_google_grpc/internal/transport/http2_server.go:336 +0x1db
created by google.golang.org/grpc/internal/transport.NewServerTransport in goroutine 174381
	google.golang.org/grpc/internal/transport/external/org_golang_google_grpc/internal/transport/http2_server.go:333 +0x2feb

goroutine 174273 [semacquire, 75 minutes]:
sync.runtime_Semacquire(0xc00a030290?)
	GOROOT/src/runtime/sema.go:62 +0x25
sync.(*WaitGroup).Wait(0xc00a030288)
	GOROOT/src/sync/waitgroup.go:116 +0xa5
golang.org/x/sync/errgroup.(*Group).Wait(0xc00a030280)
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:56 +0x38
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.Wait({0xc00a030280?, {0xe2743d0?, 0xc009e49590?}})
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:144 +0x65
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GoAndWait({0xe274408, 0xc009ef5200}, {0xc00be2bb58, 0x2, 0x9991160?})
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:192 +0x199
github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed.(*RangeFeed).run(0xc008fa7c80, {0xe274408, 0xc009ef5200}, {0xe2c7500, 0xc009e493b0})
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed/rangefeed.go:343 +0xfb1
github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed.(*RangeFeed).start.func2({0xe274408, 0xc009ef4cc0})
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed/rangefeed.go:242 +0x576
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:485 +0x263
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 173997
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExplainRedact.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-38644

--- a/pkg/ccl/testccl/sqlccl/explain_test.go
+++ b/pkg/ccl/testccl/sqlccl/explain_test.go
@@ -38,6 +38,7 @@ func TestExplainRedactDDL(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	skip.UnderDeadlock(t, "the test is too slow")
+	skip.UnderRace(t, "the test is too slow")
 
 	const numStatements = 10
 

--- a/pkg/sql/explain_test.go
+++ b/pkg/sql/explain_test.go
@@ -509,6 +509,7 @@ func TestExplainRedact(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	skip.UnderDeadlock(t, "the test is too slow")
+	skip.UnderRace(t, "the test is too slow")
 
 	const numStatements = 10
 

